hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -294,7 +306,8 @@ def from_typing_type(thing):
     _global_type_lookup.update(
         {
             typing.ByteString: st.binary(),
-            typing.Reversible: st.lists(st.integers()),
+            # Reversible is somehow a subclass of Hashable, so we tuplize it.
```

Comment:
maybe somewhere in `try_subclass` or something we need to actually check if a class is a concrete implementation if we are going to add it to the strategy.... does that make sense?

Reply:
That does make sense, but I think it's actually a separate problem - `issubclass(Reversible, Hashable)` returning True is IMO a bug upstream and I'm reporting it as such.

(and `try_subclass` is probably too general a place for that hack; I'm still working out where to put it)

Reply:
ugh.. yeah good catch i wrote that before realizing how wrong `issubclass(Reversible, Hashable)`.